### ISSUE TITLE
Labels for SQL data sources based on additional returned columns

### ISSUE THREAD

**Issue by FireEmerald** [grafana/grafana#36370]
**What would you like to be added**:
Grafana should treat additional columns returned by a query against a sql data source (e.g. postgres) as key/value pairs and allow them to be used as regular labels.

``` sql
SELECT
  captured AS time,
  name AS metric,
  city, -- additional key/value label {city=...}
  speed AS value
FROM car_status
```

time (captured)|metric (name)|city|value (speed)
--|--|--|--
2021-07-02T08:50:00|car1|london|100
2021-07-02T08:50:00|car2|berlin|150

**Why is this needed**:
Some data sources (e.g. prometheus) provide labels out-of-the-box and they are handled by grafana as-it. Sql data sources like postgres do not provide labels, you only got your query which returns columns. This becomes a **really big issue** when you try to use the new alerting system of grafana 8 - as it relies hevily on them and without labels it is kind of not useable.

References pr #32257, bug #32899.

**Comment by kylebrandt** [comment]
@FireEmerald This should work in Grafana 8. Is the format option set to "Time Series"? If it is Table the conversion to labels does not happen.

**Comment by kylebrandt** [comment]
Okay it doesn't work like that because it tries to convert the string column. Making PR that may address it, bit confused on aspects of this though.

**Comment by kylebrandt** [comment]
**Aside**:
Another option that should work with the new alerting is not return time series data from SQL (Although the information available to the alert may be lacking in this case currently). This can have multiple string columns.

You can still filter a time range, but the time column is not selected in the query, and a non-zero value is "alerting". See https://grafana.com/docs/grafana/latest/alerting/unified-alerting/grafana-managed-numeric-rule/  , in this case "Format As" is set to "Table"

**Comment by FireEmerald** [comment]
@kylebrandt Big thanks for your answers so far. Your "workaround" from the linked docu page does seem to work. 

Do you know the state of the new alerting of grafana 8? I know it must be enabled using the feature switch but it feels really like a alpha or beta. The documentation lacks so many details and there are issues in the interface.

Just for example, this switch is always disabled on my alerts:
![switch](https://user-images.githubusercontent.com/1107939/125069652-479c6180-e0b7-11eb-9c76-a5aeb814e124.png)

Or the shown `Matching instances` view shows garbage (last entry is sometimes shown why?):
![2021-07-09 13_40_12-Window](https://user-images.githubusercontent.com/1107939/125072731-3bb29e80-e0bb-11eb-9e83-1ad1b784ffcd.png)

Also the notifications send by grafana if an alert triggers are a waay delayed.

**Comment by kylebrandt** [comment]
> Do you know the state of the new alerting of grafana 8? I know it must be enabled using the feature switch but it feels really like a alpha or beta. The documentation lacks so many details and there are issues in the interface.

Currently considered a "feature preview", I think it will stay that way until _at least_ 8.2. 

> Just for example, this switch is always disabled on my alerts:
> ![switch](https://user-images.githubusercontent.com/1107939/125069652-479c6180-e0b7-11eb-9c76-a5aeb814e124.png)

Yes, this "switch" is misleading. It isn't actually a switch, it only shows / hide the options below, we have made an issue for this https://github.com/grafana/grafana/issues/36016

> Or the shown `Matching instances` view shows garbage (last entry is sometimes shown why?):
> ![2021-07-09 13_40_12-Window](https://user-images.githubusercontent.com/1107939/125072731-3bb29e80-e0bb-11eb-9e83-1ad1b784ffcd.png)
> 
> Also the notifications send by grafana if an alert triggers are a waay delayed.

Not sure what might have caused this one, can you make a new issue for it? That helps us keep track of things and make sure we are not missing stuff - as these are things that will generally need to be fixed before we can take it out of "Feature preview". 

And thank you for all the time you are investing in this feature and reporting issues, very much appreciated by the whole group working on this!
